vcd/resource_vcd_nsxt_edgegateway_dns.go

```diff
@@ -0,0 +1,384 @@
+			},
+			"snat_rule_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "The value is `true` if a SNAT rule exists for the DNS forwarder.",
+			},
+			"snat_rule_ip_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "The external IP address of the SNAT rule. " +
+					"Can be modified only if the Edge Gateway's external network is using IP spaces. (VCD 10.5.0+)",
+			},
+			"default_forwarder_zone": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: "The default forwarder zone.",
+				Elem:        defaultForwarderZone,
+			},
+			"conditional_forwarder_zone": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				MaxItems:    5,
+				Description: "Conditional forwarder zones",
+				Elem:        conditionalForwarderZone,
+			},
+		},
+	}
+}
+
+var defaultForwarderZone = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique ID of the forwarder zone.",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the forwarder zone.",
+		},
+		"upstream_servers": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MaxItems:    3,
+			Description: "Servers to which DNS requests should be forwarded to.",
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.IsIPAddress,
+			},
+		},
+	},
+}
+
+var conditionalForwarderZone = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique ID of the forwarder zone.",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the forwarder zone.",
+		},
+		"upstream_servers": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MaxItems:    3,
+			Description: "Servers to which DNS requests should be forwarded to.",
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.IsIPAddress,
+			},
+		},
+		"domain_names": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			Description: "Set of domain names on which conditional forwarding is based.",
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	},
+}
+
+func resourceVcdNsxtEdgegatewayDnsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceVcdNsxtEdgegatewayDnsCreateUpdate(ctx, d, meta, "create")
+}
+
+func resourceVcdNsxtEdgegatewayDnsUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceVcdNsxtEdgegatewayDnsCreateUpdate(ctx, d, meta, "update")
+}
+
+func resourceVcdNsxtEdgegatewayDnsCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}, origin string) diag.Diagnostics {
```

Comment:
Are you doing any partial matching as discussed earlier (I can't find that)?
If not, it would be good to add a comment/TODO (in a relevant line) about potential need of that + what the user is losing without it currently.

Reply:
Added on line 318 of the resource file.